src/grafana_service.ts

```diff
@@ -54,17 +41,27 @@ export async function queryByMetric(
   return data;
 }
 
-async function queryGrafana(metric: Metric, url: string, apiKey: string, params: any) {
+async function queryGrafana(query: MetricQuery, apiKey: string) {
   let headers = { Authorization: `Bearer ${apiKey}` };
+  let axios_query = {
+    headers,
+    url: query.url,
+    method: query.method,
+  };
+
+  _.forOwn(query.schema, (val,key) => {
+    axios_query[key] = val;
+  });
 
   try {
-    var res = await axios.get(url, { params, headers });
+    var res = await axios(axios_query);
   } catch (e) {
+    console.log(`Data kit: got response ${e.response.status}, message: ${e.message}`);
```

Comment:
Can you guarantee that `e` always has `response` here? Lets define what specific type of exception you are catching.

Reply:
`axios()` throw AxiosError if got exception. As described in type declaration it contain `response`. Let's leave it as is.